src/main/java/com/radixdlt/crypto/HashUtils.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.crypto;
+
+import com.google.common.hash.HashCode;
+import com.google.common.primitives.UnsignedBytes;
+import com.radixdlt.SecurityCritical;
+import com.radixdlt.SecurityCritical.SecurityKind;
+
+import java.security.SecureRandom;
+import java.util.Comparator;
+
+@SecurityCritical(SecurityKind.HASHING)
+public final class HashUtils {
+
+	private static final Comparator<HashCode> hashComparator = new Comparator<HashCode>() {
+
+		private final Comparator<byte[]> bytesComparator = UnsignedBytes.lexicographicalComparator();
+
+		@Override
+		public int compare(HashCode o1, HashCode o2) {
+			return bytesComparator.compare(o1.asBytes(), o2.asBytes());
+		}
+	};
+
+	private static final SecureRandom secureRandom = new SecureRandom();
+
+	private static final HashHandler shaHashHandler = new SHAHashHandler();
+
+	/**
+	 * Returns a hash consisting of 32 zeros.
+	 */
+	public static HashCode zero256() {
+		return zero(32);
+	}
+
+	/**
+	 * Returns a hash consisting of `length` zeros.

```

Comment:
Probably should mention units here.

Also Javadoc doesn't use markdown, so probably need something like:
```
 * Returns a hash consisting of {@code length} zero bytes.
```